File: userSessionDiagrams/mermaid/session.js
const crypto = require('crypto');

class Session {
    constructor(token = null) {
        this.token = token || crypto.randomBytes(32).toString('base64url');
        this.createdAt = Date.now();
        this.expiryDuration = 3600000; // 1 hour in milliseconds
    }

    static create(user) {
        if (!user) {
            throw new Error('User cannot be null');
        }
        return new Session();
    }

    isValid() {
        const currentTime = Date.now();
        const elapsedTime = currentTime - this.createdAt;
        return elapsedTime < this.expiryDuration;
    }
}

module.exports = Session;

File: userSessionDiagrams/uml/Session.py
from typing import Optional
import secrets
import time


class Session:
    def __init__(self, token: Optional[str] = None):
        self._token = token or secrets.token_urlsafe(32)
        self._created_at = time.time()
        self._expiry_duration = 3600  # 1 hour in seconds

    @classmethod
    def create(cls, user: 'User') -> 'Session':
        if not user:
            raise ValueError("User cannot be None")
        return cls()

    def is_valid(self) -> bool:
        current_time = time.time()
        elapsed_time = current_time - self._created_at
        return elapsed_time < self._expiry_duration

    @property
    def token(self) -> str:
        return self._token